digi/xbee/firmware.py

```diff
@@ -7143,7 +7143,11 @@ def update_local_firmware(target, xml_fw_file, xbee_firmware_file=None,
         msg = "Error: %s" % exc
         raise exc
     finally:
-        finished = (target._active_update_type == NodeUpdateType.FIRMWARE)
+        finished = (
+            target._active_update_type == NodeUpdateType.FIRMWARE
+            if isinstance(target, XBeeDevice)
+            else True
```

Comment:
I'm not 100% sure if this is the right thing to claim for a direct serial-port update. Would this be better?

    else (msg == 'Success')

Reply:
I'm fine with rewriting the `finished` variable as you've put it here. The end result is exactly the same, the only difference is really whether we're checking for `isinstance(target, str)` or `isinstance(target, XBeeDevice)`.

And good catch on line 7134. I'll make that change.